pytest.ini

```diff
@@ -13,6 +13,7 @@ filterwarnings =
     ignore::UserWarning:cpuinfo,
     ignore: divide by zero encountered in log
     ignore: invalid value encountered in log
+    ignore:`np.compat`, which was used during the Python 2 to 3 transition, is deprecated
```

Comment:
I think this should be left out. We should no longer be using `np.compat` in the test suite or the code base, beyond the single test to check that a deprecation warning is emitted.

Reply:
Hi @rgommers,
It's clear to me - once I remove this exclusion tests fail in a few places as `compat` is used across codebase to some extension. 

I've got a question - If a given function (e.g. `asbytes` from `numpy.compat.py3k.py`) is used in several places (here this one is used in `defchararray.py` and `_iotools.py`) should I move this specific one to e.g. `numpy._utils` and import from there? I see that `_utils` module serves purpose of a internal miscellaneous helpers.

Reply:
There may be cases where that is needed, and then `_utils` is fine, however I suspect that for most functions in `numpy.compat` they should instead be replaced by the idiomatic Python 3 way of doing things. That compat code is written the way it is to be compatible with Python 2 and 3 at the same time - and we don't have to worry about py2 anymore.

In this case, the function is:
```python
def asbytes(s):
    if isinstance(s, bytes):
        return s
    return str(s).encode('latin1')
```

In Python 3, it should be clear whether something is a string or bytes. In most places it's probably a string already, in which case `.encode('latin1')` can be used directly.

If it's too fiddly to figure out you can be pragmatic and retain `asbytes`; it doesn't matter too much (moving it somewhere private achieves the main goal), but in most cases it should be straightforward to remove the use of the compat functions completely.

Reply:
It's all clear to me! 
I added a very slimmed file in `_utils` for functions that are still used in `np.savetxt` for converting from and to bytes (in `write_normal` method). 

I other places, where is was clear, I removed the usage of `compat` items.
For example in `np.savetxt` parameter `fmt` accepts `str` and `List[str]` according to the stub and documentation. I dropped an additional acceptance of bytes (It wasn't mentioned in docs, so I assumed it came from compatibility with Python 2)